### PR TITLE
Fix delete namespace with a namespaced broker stuck on finalizer

### LINKED ISSUE #2893: deleting a namespace with a namespaced broker makes it stuck on finalizer "unable to apply dataplane manifest"
**Describe the bug**
Deleting a namespace with a namespaced broker in it makes the Broker deletion stuck on the finalizer. 

the broker finalization is blocked on

```
unable to apply dataplane manifest: configmaps "config-kafka-broker-data-plane"
        is forbidden: unable to create new content in namespace foobar because it
        is being terminated
```

 **Expected behavior**
Namespaced broker should be able to be deleted even if its namespace is undergoing deletion.

 **To Reproduce**
1. create a namespace foobar
2. create an application using a KafkaNamespaced broker in the foobar namespace, like https://gist.github.com/maschmid/b518de4afc520c718fd9acec0cb4a5af
3. delete the foobar namespace 

```
kubectl create namespace foobar
kubectl apply -n foobar -f https://gist.githubusercontent.com/maschmid/b518de4afc520c718fd9acec0cb4a5af/raw/454c9974029e33e9a58ccf247323bd6789c14b3f/kafka-namespaced-heartbeat.yaml
kubectl wait --for=condition=ready --timeout=3m -n foobar broker/broker
kubectl wait --for=condition=ready --timeout=3m -n foobar trigger/subscription
kubectl delete namespace foobar
```

 **Knative release version**
1.8

 **Additional context**
running on OpenShift, knative deployed via Serverless operator from https://github.com/openshift-knative/serverless-operator/tree/release-1.27

### PR DESCRIPTION
**Pull request by pierDipi**
When the system namespace is deleted while we're running there is no point in
trying to delete the resource from the ConfigMap since the entire ConfigMap
is gone.

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

Fixes #2893

### CODE REVIEW

File: control-plane/pkg/reconciler/broker/broker.go [hunk 1]
```diff
@@ -292,37 +292,7 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, broker *eventing.Broker)
...
-	// to update anyway remaining annotations with the contract generation that was saved in the CM.
-	// Note: if there aren't changes to be done at the pod annotation level, we just skip the update.
-
-	// Update volume generation annotation of receiver pods
-	if err := r.UpdateReceiverPodsAnnotation(ctx, logger, ct.Generation); err != nil {
-		return err
-	}
-	// Update volume generation annotation of dispatcher pods
-	if err := r.UpdateDispatcherPodsAnnotation(ctx, logger, ct.Generation); err != nil {
+	if err := r.deleteResourceFromContractConfigMap(ctx, logger, broker); err != nil {
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/broker/broker.go:295]: note: this function extraction is not a refactoring but it allows us to continue finalization while skipping ConfigMap changes as the comment on `apierrors.IsForbidden(err)` describes

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**